Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/EventParser.java

```diff
@@ -151,42 +164,58 @@ else if (event instanceof RelationEventType) {
                 link.setLabel("has object");
                 links.add(link);
             }
-            
+
             return predNode;
         }
-        
+
         return null;
     }
-    
-    private PredicateNode parsePredicateEvent(AppellationEventType appellationEvent, String statementId) {
+
+    private PredicateNode parsePredicateEvent(
+            AppellationEventType appellationEvent, String statementId) {
         PredicateNode predNode = new PredicateNode();
-        
         parseNode(appellationEvent, predNode, statementId);
-        
         predNode.setId(UUID.randomUUID().toString());
-        
         return predNode;
     }
-    
-    private void parseNode(AppellationEventType event, Node node, String statementId) {
+
+    private void parseNode(AppellationEventType event, Node node,
+            String statementId) {
         StringBuffer label = new StringBuffer();
         for (TermType type : event.getTerms()) {
             label.append(type.getTermInterpertation());
             label.append(" ");
         }
         node.setId(event.getAppellationEventID());
-       
         node.setConceptId(label.toString().trim());
-        
         if (node.getConceptId() != null) {
-            node.setLabel(conceptCollectionManager.getConceptLemmaFromConceptId(node.getConceptId()));
+            String id = node.getConceptId();
+            ConceptpowerReply re = conceptPowerConnector.getById(id);
+            node.setLabel(getLemma(re, id));
+            node.setDescription(getDesc(re, id));
         }
-        
         node.getStatementIds().add(statementId);
     }
-    
-    private ElementEventsType getElementEventTypeFromCreationEventTypeID(String relationEventId)
-            throws JAXBException, QStoreStorageException {
+
+    private String getLemma(ConceptpowerReply re, String id) {
+        String lemma = id;
```

Comment:
this logic should not be done here, but where you call this method. It goes against the "principle of least astonishment". Why would I get the id when I ask for the lemma.